### ISSUE TITLE
Terminator blank when launched from run prompt and in max

### ISSUE THREAD

**Issue by bdc34** [awesomeWM/awesome#1357]
Output of `awesome --version`:
```
awesome v3.5.2-2209-g9f2c471 (The Fox)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✘
 • LGI version: 0.9.0
```

cat /etc/lsb-release
```
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=17.3
DISTRIB_CODENAME=rosa
DISTRIB_DESCRIPTION="Linux Mint 17.3 Rosa"
```
**How to reproduce the issue:**
(first clear out personal lua startup, and Terminator .config/terminator/conf )
1. login with Mate/mdm to awesome
2. switch to max mode with modkey+space
3. hit modkey+r, type terminator, hit return
The window is blank. It is black, it has the title, and decorations, just no prompt text. Typing exit will close the window. Expected window with terminator where text is displayed.

4. switch to floating 
5. hit modkey+r, type terminator, hit return
The prompt in terminator is fine.

6. exit, and log back in
7. switch to max
8. hit modkey+return to get xterm
9. type terminator and hit return
Terminator is not black/empty.

**Other notes**

NOTE 1: The steps to reproduce are with one monitor. If I'm running with two monitors, moving the blank terminator window to the other monitor and back sometimes fixes the problem. 

NOTE 2: Often, after doing NOTE 1 the problem no longer occurs on the next time terminator is run with modkey+r 

NOTE 3: This happens for me with --no-argb or without.

NOTE 4: Nothing like this happens with Mate or Cinnamon. 

NOTE 5: .xsession-errors:
```
$ cat .xsession-errors 
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
/etc/mdm/Xsession: Beginning session setup...
localuser:bdc34 being added to access control list
2017-01-03 13:38:33 W: Cannot reliably detect EOF on an GIOInputStream with this LGI version
2017-01-03 13:38:33 ...usr/local/share/awesome/lib/beautiful/xresources.lua:54: W: Failed to get xrdb value 'background'.
2017-01-03 13:38:33 W: beautiful: can't get colorscheme from xrdb (using fallback).
2017-01-03 13:38:39 W: awesome: a_glib_poll:387: Last main loop iteration took 1.351697 seconds! Increasing limit for this warning to that value.
2017-01-03 13:38:57 W: awesome: xerror:989: X error: request=ConfigureWindow (major 12, minor 0), error=BadValue (2)
2017-01-03 13:38:57 W: awesome: xerror:989: X error: request=ConfigureWindow (major 12, minor 0), error=BadValue (2)

(emacs:26168): Gtk-CRITICAL **: gtk_distribute_natural_allocation: assertion 'extra_space >= 0' failed

(emacs:26168): Gtk-CRITICAL **: gtk_distribute_natural_allocation: assertion 'extra_space >= 0' failed
PluginRegistry::load_plugins: Importing plugin __init__.py failed: 'module' object has no attribute 'AVAILABLE'`
```

**Comment by psychon** [comment]
> 2017-01-03 13:38:57 W: awesome: xerror:989: X error: request=ConfigureWindow (major 12, minor 0), error=BadValue (2)

Huh?

Can you run awesome under xtrace (sometimes also called x11trace): https://xtrace.alioth.debian.org/

I recommend using something like `xtrace -o /tmp/something.txt awesome` somewhere. Then, reproduce the problem (and please do as few other stuff as possible) and quit awesome. Show the resulting file.

**Comment by bdc34** [comment]
I ran it under Xepher, startting xterm in the Xepher window, then running xtrace+awesome from the xterm.

https://drive.google.com/file/d/0B3MKf1DDkhjiVy1PRnBIT09DWG8/view?usp=sharing

**Comment by psychon** [comment]
Ah, Xephyr is a good idea. Thanks. :-)

> https://drive.google.com/file/d/0B3MKf1DDkhjiVy1PRnBIT09DWG8/view?usp=sharing

Did awesome say something like the following again? I cannot find anything like it in the log.

> 2017-01-03 13:38:57 W: awesome: xerror:989: X error: request=ConfigureWindow (major 12, minor 0), error=BadValue (2)

**Comment by bdc34** [comment]
The error at 2017-01-03 13:38:57 was from the .xsession-error log, not xtrace. I do not see it in the xtrace either. I don't know where the xephyr errors go. I don't seem them on stdout.

Here is a secoond xtrace.log. To help locate where the problem is starting I added a comment that can be found by searching for * just before I do modkey+r term TAB RET. 

https://drive.google.com/open?id=0B3MKf1DDkhjiVy1PRnBIT09DWG8

**Comment by p-himik** [comment]
I have a very similar problem, only with Awesome 4.0:
```
awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.2.4 (running with Lua 5.2)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✘
 • LGI version: 0.9.0
```
Also, when I resize a floating Terminator window, especially if I drag by the top left corner, the window starts shrinking and trembling like crazy.

After I switched "Window geometry hints" off in the Terminator preferences, everything went back to normal.

Previously I had Awesome v3.5 (built from the tag version) - there was no such problem, neither with the contents nor with the size.

**Comment by psychon** [comment]
> Also, when I resize a floating Terminator window, especially if I drag by the top left corner, the window starts shrinking and trembling like crazy.

Without looking at the details, I am sure that:
- You have size hints disabled in your config
- Terminator uses libvte
- $SOMETHING (I think it was libvte) is not compliant to ICCCM (You must not resize your window in a response to the WM resizing your window).

> After I switched "Window geometry hints" off in the Terminator preferences, everything went back to normal.

Yup, sounds like the above. (No more size hints to ignore)

----

Could someone try if https://github.com/awesomeWM/awesome/commit/162685bf3013e2e26ce517484014a4dabfc83927 fixes this problem?

**Comment by bdc34** [comment]
Thanks for looking into this. The patch didn't fix the problem.

I have two logs from xtrace.

1st: I only did the steps to reproduce above in Xepher/xtrace
https://drive.google.com/open?id=0B3MKf1DDkhjiX2w3TUFaNXdzQnc

2nd: I did the steps to reproduce but did mod+r->terminator several times. It worked fine and was not blank the first, and third time. It was blank the second, fourth and fifth time.
https://drive.google.com/open?id=0B3MKf1DDkhjid3REMi1WbG5wWjA

I do not get a trembling terminator window in either version.

Also, my Xephyr steps are
1. from my normal awesome session open terminal type `Xephyr :3 -ac -br -noreset -screen 1152x720`
2. Then `xtrace -D:9 -d:3 -o ~/xtrace.patch.log -n`
3. Then `DISPLAY=:3.0 xterm`
4. Move mouse to xterm in Xephyr window, hit ctrl+shift
5. In that xterm type `DISPLAY=:9.0 awesome --no-argb -c ~/src/awesome/awesomerc.lua`
6. Then do the steps to reproduce.

Version:
```
awesome v4.0-84-g162685b (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✘
 • LGI version: 0.9.0
```
